src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -111748,6 +111748,140 @@ static void M2(string s1, string s2)
+                // (18,14): error CS1501: No overload for method 'Equals' takes 0 arguments
+                //         this.Equals();
+                Diagnostic(ErrorCode.ERR_BadArgCount, "Equals").WithArguments("Equals", "0").WithLocation(18, 14),
+                // (19,9): error CS0176: Member 'object.Equals(object, object)' cannot be accessed with an instance reference; qualify it with a type name instead
+                //         this.Equals(null, null);
+                Diagnostic(ErrorCode.ERR_ObjectProhibited, "this.Equals").WithArguments("object.Equals(object, object)").WithLocation(19, 9));
+        }
+
+        [Fact]
+        [WorkItem(35816, "https://github.com/dotnet/roslyn/issues/35816")]
+        public void EqualsMethod_DefaultArgument_01()
+        {
+            var source = @"
+#nullable enable
+class C // 1
+{
+    static void M(C c1)
+    {
+        if (c1.Equals())
+        {
+            // technically the arg to Equals was 'null' so this should be analyzed as maybe-null
+            // however this scenario seems like such an anti-pattern that we shouldn't be concerned about it.
+            _ = c1.ToString();
+        }
+
+        if (c1.Equals(null))
+        {
+            _ = c1.ToString(); // 2
+        }
+    }
+
+    public override bool Equals(object? other = null) => false;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (3,7): warning CS0659: 'C' overrides Object.Equals(object o) but does not override Object.GetHashCode()
+                // class C // 1
+                Diagnostic(ErrorCode.WRN_EqualsWithoutGetHashCode, "C").WithArguments("C").WithLocation(3, 7),
+                // (16,17): warning CS8602: Dereference of a possibly null reference.
+                //             _ = c1.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1").WithLocation(16, 17));
+        }
+
+        [Fact]
+        [WorkItem(35816, "https://github.com/dotnet/roslyn/issues/35816")]
+        public void EqualsMethod_DefaultArgument_02()
+        {
+            var source = @"
+#nullable enable
+class C : System.IEquatable<C>
+{
+    static void M(C c1)
+    {
+        if (c1.Equals())
+        {
+            // technically the arg to Equals was 'null' so this should be analyzed as maybe-null
+            // however this scenario seems like such an anti-pattern that we shouldn't be concerned about it.
+            _ = c1.ToString();
+        }
+
+        if (c1.Equals(null))
+        {
+            _ = c1.ToString(); // 1
+        }
+    }
+
+    public bool Equals(C? other = null) => false;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (16,17): warning CS8602: Dereference of a possibly null reference.
+                //             _ = c1.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c1").WithLocation(16, 17));
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider adding a simple test that matches the bug report. Perhaps using the SemanticModel with:
```C#
class Program
{
    static void Main()
    {
        ReferenceEquals(
    }
}
```

Reply:
Note that when VS crashes, the currently displayed text has a closing parenthesis. It’s added automatically by VS; it’s not typed by the user, which is why I omitted it from my repro steps.